### PR TITLE
fix: Nested Dropdown components

### PR DESCRIPTION
**Pull request by koba04**
## Related URL

Fix #921

## Overview

Fix a problem for nested Dropdown components.
See the RTA story of Dropdown Component in storybook.

## What I did

6ccbd92 is a solution using `className`, so it requires to add a `className` into Portal elements as @wmoai mentioned. We can add the `className` into Portal elements created by `smarthr-ui`, but we cannot add the `className` into Portal elements created in outside `smarthr-ui`, so it is a limited solution, but it can fixes some basic cases like nest Dropdown components.

0eb3336 is another solution for this. This uses event delegation mode in DOM and works only in React v17. You can see the change in React at the blog post.
https://reactjs.org/blog/2020/08/10/react-v17-rc.html#changes-to-event-delegation

This works fine with Portal elements not only components in `smarthr-ui`, but it works only in React v17 (BREAKING CHANGE).

I guess the two solutions have the same behavior in the case of nested Dropdown components. So we can fix by the `className` solution first (revert 0eb3336 and 872072a), and apply the fix using event delegation model later with a BREAKING CHANGE.

### CODE REVIEW

File: src/components/Dropdown/Dropdown.stories.tsx [hunk 1]
```diff
@@ -189,6 +189,31 @@ storiesOf('Dropdown', module)
       </Legends>
     </Wrapper>
   ))
+  .add('rta', () => (
```
Comment (wmoai) [inline src/components/Dropdown/Dropdown.stories.tsx:192]: I'm sorry! This change of the stories is a sample for the problem and does not need for master, so please revert my commit (b3a5aebc0b1182dee635bf54797f7841b0639fd3).

**Review by wmoai — CHANGES_REQUESTED** [review]

File: src/components/Dropdown/Dropdown.stories.tsx (hunk 1 above)
Comment (koba04) [reply src/components/Dropdown/Dropdown.stories.tsx]: @wmoai Thank you! I've reverted the commit 👍

**Review by wmoai — DISMISSED** [review]
Thanks, LGTM!

**Review by nabeliwo — APPROVED** [review]
LGTM 🎉

**Comment by koba04** [comment]
fixes via #1003

**Pull request closed without merging**